File: android-native/app/src/main/java/com/daidai/panel/data/model/ApiResponse.kt
package com.daidai.panel.data.model

import com.google.gson.annotations.SerializedName

data class ApiResponse<T>(
    @SerializedName("code") val code: Int = 0,
    @SerializedName("message") val message: String = "",
    @SerializedName("data") val data: T? = null
) {
    val isSuccess: Boolean get() = code == 0 || code == 200
}

data class PaginatedData<T>(
    @SerializedName("items") val items: List<T> = emptyList(),
    @SerializedName("total") val total: Int = 0
)

File: ios-native/DaidaiPanel/data/model/ApiResponse.swift
import Foundation

struct ApiResponse<T: Codable>: Codable {
    let code: Int
    let message: String
    let data: T?

    var isSuccess: Bool { code == 200 }

    enum CodingKeys: String, CodingKey {
        case code, message, data
    }
}

struct PaginatedData<T: Codable>: Codable {
    let items: [T]
    let total: Int

    var isEmpty: Bool { items.isEmpty }
    var count: Int { items.count }

    enum CodingKeys: String, CodingKey {
        case items, total
    }
}

struct EmptyData: Codable {}
